### PR TITLE
Adding in a Jupyter notebook example for daal4py

### PR DESCRIPTION
**Pull request by triskadecaepyon**
Adds a jupyter notebook example for daal4py.  Uses daal4py without scikit-learn and a custom dataset from the pyworkout-toolkit examples made to csv.

### CODE REVIEW

**Comment by triskadecaepyon** [comment]
There will be a corresponding article for this on Sphinx, which is why it is light on it in the notebook (notebook will be a lightly annotated follow along).

- I will adjust the train/test split, just note I have been avoiding it because it will use scikit-learn for the implementation or a hard to read pandas implementation.

- _Printing the number of features of the trained model is not really meaningful._ -- If the model is migrated via pickle to another system it would be a useful validation step or during initial setup.

- I will fix the other typos and naming.

**Comment by triskadecaepyon** [comment]
I've made changes to the main page (including content), and I've also created an example page to house the regular examples in addition to the jupyter notebook example.  Please let me know if you need any expansions or changes.

File: doc/index.rst [hunk 1]
```diff
@@ -7,14 +7,62 @@ Fast, Scalable and Easy Machine Learning With DAAL4PY
...
 ---------
-A daal4py machine learning algorithm gets constructed with a rich set of
-parameters. Let's assume we want to find the initial set of centroids for kmeans.
-We create an algorithm and configure it for 10 clusters using the 'PlusPlus' method::
+daal4py was created to give data scientists the easist way to utilize Intel® DAAL's
+powerful machine learning building blocks directly in a high-productivity manner.
+A simplified API gives high-level abstractions to the user with minimal boilerplate,
+allowing for quick to write and easy to maintain code when utilizing Jupyter Notebooks.
+For scaling capabilities, daal4py also provides the ability to do distributed machine
+learning, giving a quick way to scale out without having to manually write the
```
Comment (fschlimb) [inline doc/index.rst]: Don't compare it to anything in the DAAL world, e.g. that wiring is needed. Just say it's super-easy to scale out.

File: doc/index.rst [hunk 2]
```diff
@@ -7,14 +7,62 @@ Fast, Scalable and Easy Machine Learning With DAAL4PY
...
+
+For framework designers, daal4py's has been fashioned to be built under other
+frameworks from both an API and feature perspective.  The machine learning models split
+the training and inference classes, allowing the model to be exported and serialized
+if desired.  This design also gives the flexibility to work directly with the model and
+associated primitives, allowing one to customize the behavior of the model itself.
+The daal4py package can be built with customized algorithm loadouts, allowing for a
+smaller footprint of dependencies when neessary.
+
+Daal4py's Design
```
Comment (fschlimb) [inline doc/index.rst]: Move this to after the simple code-examples.

File: doc/index.rst [hunk 3]
```diff
@@ -7,14 +7,62 @@ Fast, Scalable and Easy Machine Learning With DAAL4PY
...
+DAAL performance in a flexible design to Data Scientists and Framework designers.
+The package uses Jinja templates to generate Cython-wrapped DAAL C++ headers, with
+Cython as a bridge between the generated DAAL code and the Python layer.
+This design allows for quicker development cycles and acts as a reference design
+to those looking to tailor their build of daal4py.  Cython also allows for better
+Python behavior, both for compatability to different frameworks and for
+pickling and serialization.
+
+API Design and usage
+---------
```
Comment (fschlimb) [inline doc/index.rst]: All underlines should have same size as headings, getting tons of warnings otherwise.

File: doc/index.rst [hunk 4]
```diff
@@ -7,14 +7,62 @@ Fast, Scalable and Easy Machine Learning With DAAL4PY
...
+smaller footprint of dependencies when neessary.
+
+Daal4py's Design
+---------
+The design of daal4py utilizes several different technologies to deliver Intel®
+DAAL performance in a flexible design to Data Scientists and Framework designers.
+The package uses Jinja templates to generate Cython-wrapped DAAL C++ headers, with
+Cython as a bridge between the generated DAAL code and the Python layer.
+This design allows for quicker development cycles and acts as a reference design
+to those looking to tailor their build of daal4py.  Cython also allows for better
```
Comment (fschlimb) [inline doc/index.rst]: better than what? Just say its good.

File: doc/index.rst [hunk 5]
```diff
@@ -32,15 +80,15 @@ The full example could look like this::
...
+One can even run this on a cluster by simple adding initializing/finalizing the
 network and adding a keyword-parameter::
 
     from daal4py import daalinit, daalfini, kmeans_init
     daalinit()
     kmeans_init(10, method="plusPlusDense", distributed=True).compute(list_of_files)
     daalfini()
 
-It's Fast
+Built for Performance
```
Comment (fschlimb) [inline doc/index.rst]: We need single-node performance charts

File: doc/examples.rst [hunk 6]
```diff
@@ -0,0 +1,145 @@
...
+
+- `Linear Regression <https://github.com/IntelPython/daal4py/blob/master/examples/daal4py_data_science.ipynb>`_
+
+General usage
+---------
+
+Principal Component Analysis (PCA) Transform
+
+- `Single-Process PCA <https://github.com/IntelPython/daal4py/blob/master/examples/pca_batch.py>`_
+- `Multi-Process (DSPV) PCA <https://github.com/IntelPython/daal4py/blob/master/examples/pca_dspv.py>`_
```
Comment (fschlimb) [inline doc/examples.rst]: Please remove all DSPV examples.

**Pull request merged**